ReadbasedAnalysis.wdl

```diff
@@ -217,94 +225,102 @@ task finish_reads {
-    Boolean enabled_tools_gottcha2
-    Boolean enabled_tools_kraken2
-    Boolean enabled_tools_centrifuge
-    String db_gottcha2
-    String db_kraken2
-    String db_centrifuge
-    String docker
-    File? gottcha2_report_tsv
-    File? gottcha2_info
-    File? centrifuge_report_tsv
-    File? centrifuge_info
-    File? kraken2_report_tsv
-    File? kraken2_info
-    String info_filename = "profiler.info"
-
+    input {
+        Boolean enabled_tools_gottcha2
+        Boolean enabled_tools_kraken2
+        Boolean enabled_tools_centrifuge
+        String db_gottcha2
+        String db_kraken2
+        String db_centrifuge
+        String docker
+        File gottcha2_report_tsv
+        File gottcha2_info
+        File centrifuge_report_tsv
+        File centrifuge_info
+        File kraken2_report_tsv
+        File kraken2_info
+        String info_filename = "profiler.info"
+    }
     command <<<
-        set -euo pipefail
 
+        set -euo pipefail
         # generate output info file
 
         info_text="Taxonomy profiling tools and databases used: "
-        echo $info_text > ${info_filename}
+        echo $info_text > ~{info_filename}
 
-        if [[ ${enabled_tools_kraken2} == true ]]
+                echo $info_text > ~{info_filename}
+
+        if [[ ~{enabled_tools_kraken2} == true ]]
         then
-            software_ver=`cat ${kraken2_info}`
-            #db_ver=`echo "${db_kraken2}" | rev | cut -d'/' -f 1 | rev`
-            db_ver=`cat ${db_kraken2}/db_ver.info`
+            software_ver=`cat ~{kraken2_info}`
+            #db_ver=`echo "~{db_kraken2}" | rev | cut -d'/' -f 1 | rev`
+            db_ver=`cat ~{db_kraken2}/db_ver.info`
             info_text="Kraken2 v$software_ver (database version: $db_ver)"
-            echo $info_text >> ${info_filename}
+            echo $info_text >> ~{info_filename}
         fi
+        echo $info_text > ~{info_filename}
 
-        if [[ ${enabled_tools_centrifuge} == true ]]
+        if [[ ~{enabled_tools_centrifuge} == true ]]
         then
-            software_ver=`cat ${centrifuge_info}`
-            db_ver=`cat $(dirname ${db_centrifuge})/db_ver.info`
+            software_ver=`cat ~{centrifuge_info}`
+            db_ver=`cat $(dirname ~{db_centrifuge})/db_ver.info`
             info_text="Centrifuge v$software_ver (database version: $db_ver)"
-            echo $info_text >> ${info_filename}
+            echo $info_text >> ~{info_filename}
         fi
 
-        if [[ ${enabled_tools_gottcha2} == true ]]
+        if [[ ~{enabled_tools_gottcha2} == true ]]
         then
-            software_ver=`cat ${gottcha2_info}`
-            db_ver=`cat $(dirname ${db_gottcha2})/db_ver.info`
+            software_ver=`cat ~{gottcha2_info}`
+            db_ver=`cat $(dirname ~{db_gottcha2})/db_ver.info`
             info_text="Gottcha2 v$software_ver (database version: $db_ver)"
-            echo $info_text >> ${info_filename}
+            echo $info_text >> $~info_filename}
```

Comment:
having both the $ & ~ here i believe is creating an issue where the gottcha2 version info is not being populated to the info file. It should just be  the '~'